jobs/build/ose/scripts/merge-and-build.sh

```diff
@@ -181,44 +180,38 @@ cd ose
 # https://github.com/openshift/ose/commit/02b57ed38d94ba1d28b9bc8bd8abcb6590013b7c
 git config merge.ours.driver true
 
-# Set to empty string to use tito's normal version progression
-export TITO_USE_VERSION=""
-
-if [ "${BUILD_MODE}" == "enterprise" ]; then
-
-  git checkout -q enterprise-${OSE_VERSION}
+if [ "${BUILD_MODE}" == "online:stg" ] ; then
+  CURRENT_BRANCH="stage"
+  UPSTREAM_BRANCH="upstream/stage"
+elif [ "${BUILD_MODE}" == "enterprise:pre-release" ] ; then
+  CURRENT_BRANCH="enterprise-${OSE_VERSION}"
+  UPSTREAM_BRANCH="upstream/release-${OSE_VERSION}"
+elif [ "${BUILD_MODE}" == "enterprise" ]; then
+  CURRENT_BRANCH="enterprise-${OSE_VERSION}"
+  # UPSTREAM_BRANCH is not needed in this case
+else # Otherwise, online:int
+  CURRENT_BRANCH="master"
+  UPSTREAM_BRANCH="upstream/master"
+fi
 
-else
+echo "Building from branch: ${CURRENT_BRANCH}"
+git checkout -q ${CURRENT_BRANCH}
 
+if [ "${BUILD_MODE}" != "enterprise" ]; then
   # If we are here, we are building master or stage for online
-
-  # Creating a target version allows online:int builds to resume where the last stage build left off in terms
-  # of versioning. This should not be necessary when we can safely use a different 'release' in the tito version.
-  export TITO_USE_VERSION="--use-version=$(get_post_stage_version origin.spec)"
```

Comment:
This was done here on purposes. Once upstream is added, `get_post_stage_version` can no longer checkout "stage" . `get_post_stage_version` can be changed to disambiguate if you want to keep the refactor.